backend/tests/test_endpoints.py

```diff
@@ -32,4 +47,108 @@ async def test_listings_pagination_with_limit(client, load_payload):
+    
+    response = requests.post("http://127.0.0.1:8000/login", json=login_data)
+    
+    assert response.status_code == 400
+    data = response.json()
+    assert "detail" in data
+    assert data["detail"] == "Incorrect email or password"
+
+@pytest.mark.asyncio
+async def test_login_missing_password_422(client, load_payload):
+    login_payloads = load_test_payload("login_payloads.json")
+    login_data = login_payloads["missing_password"]
+    
+    response = requests.post("http://127.0.0.1:8000/login", json=login_data)
+    
+    assert response.status_code == 422
+    data = response.json()
+    assert "detail" in data
+
+# ================================== SIGNUP TESTS ==================================
+@pytest.mark.asyncio
+async def test_signup_success(client, load_payload):
+    signup_payloads = load_test_payload("signup_payloads.json")
+    signup_data = signup_payloads["valid_signup"]
+    test_email = signup_data["email"]
+    
+    try:
+        response = requests.post("http://127.0.0.1:8000/sign-up", json=signup_data)
+        
+        assert response.status_code == 201
+        data = response.json()
+        assert "user_id" in data
+        assert "message" in data
+        assert data["message"] == "User registered successfully."
+    finally:
+        await cleanup_test_user(test_email)
+
+@pytest.mark.asyncio
+async def test_signup_invalid_email_400(client, load_payload):
+    signup_payloads = load_test_payload("signup_payloads.json")
+    signup_data = signup_payloads["invalid_email"]
+    
+    response = requests.post("http://127.0.0.1:8000/sign-up", json=signup_data)
+    
+    assert response.status_code == 400
+    data = response.json()
+    assert "detail" in data
+    assert data["detail"] == "Invalid email format."
+
+@pytest.mark.asyncio
+async def test_signup_duplicate_email_409(client, load_payload):
+    signup_payloads = load_test_payload("signup_payloads.json")
+    signup_data = signup_payloads["duplicate_email"]
+    test_email = signup_data["email"]
+    
+    # First signup
+    response = requests.post("http://127.0.0.1:8000/sign-up", json=signup_data)
+    assert response.status_code == 201
+    
+    # Try to signup with same email
+    response = requests.post("http://127.0.0.1:8000/sign-up", json=signup_data)
+    
+    assert response.status_code == 409
+    data = response.json()
+    assert "detail" in data
+    assert data["detail"] == "Email already registered."
+
+@pytest.mark.asyncio
+async def test_signup_missing_fields_422(client, load_payload):
+    signup_payloads = load_test_payload("signup_payloads.json")
+    signup_data = signup_payloads["missing_password"]
+    
+    response = requests.post("http://127.0.0.1:8000/sign-up", json=signup_data)
+    
+    assert response.status_code == 422
+    data = response.json()
+    assert "detail" in data
+
+@pytest.mark.asyncio
+async def test_signup_server_error_500(client, load_payload):
```

Comment:
This is fine to omit. 500 errors should be reserved truly for catch-all exceptions anyway -- that is exceptions that we never expected in testing/development to begin with.